CONTRIBUTING.md

```diff
@@ -29,3 +29,138 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
+is not expected that there will be a 3.x release of the specification in the
+foreseeable future.
+
+### Compatibility and Feature Enablement
+
+For the purposes of this discussion we classify features into the following buckets:
+
+1. Backwards compatible. A file written under an older version of the format
+   should be readable under a newer version of the format.
+
+2. Forwards compatible. A file written under a newer version of the format with
+   the enabled feature can be read under an older version of the format, but
+   some information might be missing or performance might be suboptimal.
+
+3. Forward incompatible. A file written under a new version of the format with
+   the feature enabled cannot be read under an older version of the format (e.g.
+   Adding a new compression algorithm).
+
+New features are intended to be widely beneficial to users of Parquet, and
+therefore it is hoped third-party implementations will adopt them quickly after
+they are introduced. It is assumed that writing new parts of the format, and
+especially forward incompatible features, will be configured with feature flag
+defaulted to "off" and at some future point the features are turned on by default
+(reading of the new feature will typically be enabled without configuration or
+defaulted to on). Some amount of lead time is desirable to ensure a critical
+mass of Parquet implementations support a feature to avoid compatability issues
+across the ecosystem.  Therefore, the Parquet PMC gives the following
+recommendations for managing features:
+
+1. Backwards compatibility is the concern of implementations but given the
+   ubiquity of Parquet and the length of time it has been used, libraries should
+   support reading older version of the format to the greatest extent possible.
+
+2. Forward compatible features/changes may be used by default in implementations
+   once the parquet-format containing those changes has been formally released.
+   For features that may pose a significant performance regression to older
+   format readers, libaries should consider delaying default enablement until 1
+   year after the release of the parquet-java implementation that contains the
+   feature implementation.
+
+3. Forwards incompatible features/changes should not be turned on by default
+   until 2 years after the parquet-java implementation containing the feature is
+   released. It is recommended that changing the default value for a forward
+   incompatible feature flag be done as part of a major release of an
+   implementation (it is out of the scope for this guidance on how and when
+   implementations decide to do releases).
+
+For forward compatible changes which have a high chance of performance
+regression for older readers and forward incompatible changes implementations
+should clearly document the compatibility issues and should consider logging a
+warning when such a feature is used. Additionally, while it is up to maintainers
```

Comment:
Uh, warnings are generally distracting and harmful if they do not actually tell the user about something unexpected. Especially, if the user explicitly enabled the feature, they should definition *not* be shown a warning patronizing them (at least, I would strongly recommend against this in Parquet C++ or PyArrow).

Reply:
I'll remove the logging section.

Reply:
Removed.